operations.go

```diff
@@ -9,60 +9,73 @@ type operation func()
 
 // Operations is a task executor.
 type operations struct {
-	ops     []operation
-	mu      sync.Mutex
-	startMu sync.Mutex
-	done    chan struct{}
+	mu   sync.Mutex
+	busy bool
+	ops  []operation
 }
 
 func newOperations() *operations {
 	closed := make(chan struct{})
 	close(closed)
-	return &operations{
-		done: closed,
-	}
+	return &operations{}
 }
 
 // Enqueue adds a new action to be executed. If there are no actions scheduled,
 // the execution will start immediately in a new goroutine.
 func (o *operations) Enqueue(op operation) {
+	if op == nil {
+		return
+	}
+
 	o.mu.Lock()
-	defer o.mu.Unlock()
+	busy := o.busy
 	o.ops = append(o.ops, op)
-	if len(o.ops) == 1 {
-		done := make(chan struct{})
-		o.done = done
+	o.busy = true
+	o.mu.Unlock()
 
-		go func() {
-			o.startMu.Lock()
-			defer o.startMu.Unlock()
-			o.start()
-			close(done)
-		}()
+	if !busy {
+		go o.start()
 	}
 }
 
-// Done will return a channel that will be closed as soon as all currently
-// enqueued operations are finished.
-func (o *operations) Done() <-chan struct{} {
-	o.mu.Lock()
-	defer o.mu.Unlock()
-	return o.done
+// Done blocks until all currently enqueued operations are finished executing.
```

Comment:
Mind returning a `context.Context`, then the caller can give up waiting if they want. In the future might be useful for timeouts/measuring stuff.